dask_distance/_compat.py

```diff
@@ -89,3 +89,18 @@ def _indices(dimensions, dtype=int, chunks=None):
         )
 
     return grid
+
+
+def _ravel(a):
+    a = _asarray(a)
+
+    r = dask.array.atop(
+        numpy.ravel, (a.ndim + 1,),
+        a, tuple(range(0, a.ndim)),
+        dtype=a.dtype,
+        new_axes={(a.ndim + 1): a.size},
+        adjust_chunks={0: int(numpy.prod([c[0] for c in a.chunks]))},
+        concatenate=True
```

Comment:
Seems that `concatenate` is fusing everything into one big array. Will need to get away from this probably by moving to a smaller chunking first.

Reply:
Fixed in PR ( https://github.com/jakirkham/dask-distance/pull/63 ).